### PR TITLE
Publish non-repudiation client libraries to Artifactory

### LINKED ISSUE #9097: Non-repudiation: publish to Artifactory
Publish the fat JAR for the app and the JAR for the client to Artifactory.

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
changelog_begin
changelog_end

Closes #9097

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ci/publish-artifactory.sh [hunk 1]
```diff
@@ -37,6 +37,17 @@ push daml-script-runner $SCRIPT_RUNNER.asc
 push non-repudiation $NON_REPUDIATION
 push non-repudiation $NON_REPUDIATION.asc
 
+NON_REPUDIATION_CLIENT_JAR=non-repudiation-client-$RELEASE_TAG.jar
+NON_REPUDIATION_CLIENT_POM=non-repudiation-client-$RELEASE_TAG.pom
+NON_REPUDIATION_CLIENT_SRC=non-repudiation-client-$RELEASE_TAG-sources.jar
+
+push connect-ee-mvn/com/daml $NON_REPUDIATION_CLIENT_JAR
+push connect-ee-mvn/com/daml $NON_REPUDIATION_CLIENT_JAR.asc
```
Comment (stefanobaghino-da) [inline ci/publish-artifactory.sh:45]: I copied this from above, but I can't see where these files are created and I don't see them locally when I `bazel build` the relative target. Should these stay here?

Comment (cocreature) [reply ci/publish-artifactory.sh:45]: They are created here https://github.com/digital-asset/daml/blob/45c4ba2230854e048027c999d476b3b85f5393d1/azure-pipelines.yml#L92

File: ci/copy-unix-release-artifacts.sh [hunk 2]
```diff
@@ -49,4 +49,13 @@ if [[ "$NAME" == "linux" ]]; then
     NON_REPUDIATION=non-repudiation-$RELEASE_TAG.jar
     bazel build //runtime-components/non-repudiation-app:non-repudiation-app_deploy.jar
     cp bazel-bin/runtime-components/non-repudiation-app/non-repudiation-app_deploy.jar $OUTPUT_DIR/github/$NON_REPUDIATION
+
+    NON_REPUDIATION_CLIENT_JAR=non-repudiation-client-$RELEASE_TAG.jar
+    NON_REPUDIATION_CLIENT_POM=non-repudiation-client-$RELEASE_TAG.pom
+    NON_REPUDIATION_CLIENT_SRC=non-repudiation-client-$RELEASE_TAG-sources.jar
```
Comment (cocreature) [inline ci/copy-unix-release-artifacts.sh:55]: Should we include javadoc as well?

**Review by cocreature — APPROVED** [review]
Thanks!

File: ci/publish-artifactory.sh (hunk 1 above)
Comment (stefanobaghino-da) [reply ci/publish-artifactory.sh:45]: As per @cocreature's [approval](https://github.com/digital-asset/daml/pull/9282#pullrequestreview-623962794), it appears that these are safe to stay.

File: ci/copy-unix-release-artifacts.sh (hunk 2 above)
Comment (stefanobaghino-da) [reply ci/copy-unix-release-artifacts.sh:55]: Is there a specific target to run to get it? I didn't see it in the `bazel-bin` directory for `//runtime-components/non-repudiation-client`.

Comment (stefanobaghino-da) [reply ci/copy-unix-release-artifacts.sh:55]: This is the full extend of the javadoc in the library, though:

```
/**
 * A gRPC client-side interceptor that uses a key pair to sign a payload and adds it as metadata to
 * the call, alongside a fingerprint of the public key and the algorithm used to sign.
 */
 ```
 
 Maybe at this point it wouldn't make too much sense to publish it, unless it's important to have it for some reason I'm missing.

Comment (cocreature) [reply ci/copy-unix-release-artifacts.sh:55]: I know Maven mandates having it but I think artifactory doesn’t. I’d probably still publish it. It seems easier to publish javadocs everywhere than to to a per-library decision. As for why the target is not there, it looks like we need to add the following tag for this to work:
```
        "javadoc_root_packages=com.daml.nonrepudiation.client",
```

Comment (stefanobaghino-da) [reply ci/copy-unix-release-artifacts.sh:55]: Mmmh, I remember trying to add it and having issues with that, I'll see if that's still the case, thanks for the tip.

Comment (stefanobaghino-da) [reply ci/copy-unix-release-artifacts.sh:55]: Addressed by https://github.com/digital-asset/daml/pull/9282/commits/da3cc126e64751aa86902af5bc0b2833bf1c626c

**Pull request merged**